components/scream/scripts/scripts-tests

```diff
@@ -125,6 +125,22 @@ def test_branch_switch(test_obj, repo, branch_name):
     test_obj.assertEqual(get_current_branch(repo=repo), branch_name)
     test_obj.assertEqual(get_current_head(repo=repo), branch_name)
 
+###############################################################################
+def test_omp_spread(test_obj, output, num_places):
+###############################################################################
+    lines = output.splitlines()
+    all_places = []
+    for line in lines:
+        if "OMP_PLACES = '" in line:
+            places_str = line.split("=")[1].strip().strip("'")
+            places = [int(item.strip("{}")) for item in places_str.split(",")]
+            all_places.extend(places)
+
+    all_places.sort()
+    test_obj.assertEqual(len(all_places), num_places)
+    for idx, place in enumerate(all_places):
+        test_obj.assertEqual(idx, place)
```

Comment:
What if `all_places` contains duplicates? Shouldn't we make it unique first?

Edit: Ah, is it b/c when this test is run, only the `resource_spread_rank` and `resource_spread_thread` are used?

Edit2: Wait, those tests still add up to 40+40=80 threads used, which is likely to be more than the machine has, which means that my comment might still be correct.

Reply:
Also, I think this is testing that all corese 0-N are used, but it's not testing that each test spreads the threads on different cores.

Reply:
Ah, I see. Would be nice to also run a test where we ensure that separate tests are spread out. Not sure if you want to add that here though.